src/constantProductCalculator.ts

```diff
@@ -47,6 +50,71 @@ export function getConstantProductMintedLiquidityTokens(
   return ltA > ltB ? ltB : ltA;
 }
 
+export function getConstantProductZapParams(
+  liqA: bigint,
+  liqB: bigint,
+  zapAmount: bigint,
+  feeBps: bigint,
+  pactFeeBps: bigint,
+): ZapParams {
+  const swapDeposited = getSwapAmountDepositedFromZapping(
+    zapAmount,
+    liqA,
+    feeBps,
+    pactFeeBps,
+  );
+  const swapReceived = getConstantProductSwapGrossAmountReceived(
+    liqA,
+    liqB,
+    swapDeposited,
+  );
+  const primaryAddLiq = zapAmount - swapDeposited;
+  const secondaryAddLiq = getSecondaryAddedLiquidityFromZapping(
+    swapDeposited,
+    liqA,
+    liqB,
+    feeBps,
+  );
+  return { swapDeposited, swapReceived, primaryAddLiq, secondaryAddLiq };
```

Comment:
I don't think that there is any need to return `swapReceived`, the more it doesn't have fee applied, so it won't be exactly what user would get. 

I added that in research script, because I needed this for some charts.